Framework/PythonInterface/plugins/algorithms/poldi_utils.py

```diff
@@ -124,6 +129,28 @@ def get_final_dspac_array(bin_width: float, dspac_min: float, dspac_max: float,
     return np.linspace(dspac_min, dspac_max, nbins_dspac)
 
 
+def get_dspac_array_from_ws(ws: Workspace2D, lambda_min: float = 1.1, lambda_max: float = 5.0) -> np.ndarray[float]:
+    """
+    Function to calculate d-spacing bins given workspace
+    :param ws_2d: MatrixWorkspace containing POLDI data (raw instrument)
+    :param lambda_min: maximum wavelength (Ang) to consider
+    :param lambda_max: maximum wavelength (Ang) to consider
+    :return np.ndarray: array of d-spacing bins
+    """
+    _, slit_offsets, _, l1_chop = get_instrument_settings_from_log(ws)
+    # get detector positions from IDF
+    si = ws.spectrumInfo()
+    nspec = ws.getNumberHistograms()
+    tths = np.array([si.twoTheta(ispec) for ispec in range(nspec)])
+    l2s = np.asarray([si.l2(ispec) for ispec in range(nspec)])
+    l1 = si.l1()
+    # determine npulses to include in calc
+    time_max = get_max_tof_from_chopper(l1, l1_chop, l2s, tths, lambda_max) + slit_offsets[0]
+    dspac_min, dspac_max = get_dspac_limits(tths.min(), tths.max(), lambda_min, lambda_max)
```

Comment:
```suggestion
    dspac_min, dspac_max = get_dspac_limits(tths.min(), min(tths.max(), np.pi/2), lambda_min, lambda_max)
```
Maximum sin(tth) is actually at 90 degrees - will need to fix this in code called by auto-correlation as well